src/cpp/rtps/history/CacheChangePool.cpp

```diff
@@ -119,6 +122,7 @@ bool CacheChangePool::reserve_Cache(CacheChange_t** chan, uint32_t dataSize)
                 }
             }
             *chan = m_freeCaches.back();
+            m_freeCaches.pop_back();
             m_freeCaches.erase(m_freeCaches.end()-1);
```

Comment:
Either pop_back or erase, otherwise we are leaking one change every time we return one.